packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -586,153 +637,525 @@ async function handler(req: NextApiRequest & { userId?: number | undefined }) {
-      return { ...attendee, language: { translate: tAttendees, locale: language ?? "en" } };
-    });
+    if (dayjs(booking.startTime).utc().format() === evt.startTime) {
+      return booking;
+    }
 
-    evt = { ...evt, attendees: [...bookingAttendees, invitee[0]] };
+    // If rescheduling an exisiting attendee
+    if (rescheduleUid) {
+      // Check if the host or attendee is rescheduling
+      const host = booking.user?.email === req.body.email;
 
-    if (eventType.seatsPerTimeSlot <= booking.attendees.length) {
-      throw new HttpError({ statusCode: 409, message: "Booking seats are full" });
-    }
+      // If it is the host then continue rescheduling as normal
+      if (host) return;
 
-    if (booking.attendees.find((attendee) => attendee.email === invitee[0].email)) {
-      throw new HttpError({ statusCode: 409, message: "Already signed up for time slot" });
-    }
+      let seatAttendee: Partial<Person> | null = null;
+      if (!host) {
+        seatAttendee =
+          originalRescheduledBooking?.attendees.find((attendee) => attendee.email === req.body.email) || null;
+      }
 
-    const videoCallReference = booking.references.find((reference) => reference.type.includes("_video"));
+      if (!seatAttendee) {
+        throw new HttpError({ statusCode: 404, message: "Attendee not found" });
+      }
 
-    if (videoCallReference) {
-      evt.videoCallData = {
-        type: videoCallReference.type,
-        id: videoCallReference.meetingId,
-        password: videoCallReference?.meetingPassword,
-        url: videoCallReference.meetingUrl,
-      };
-    }
+      seatAttendee.language = { translate: tAttendees, locale: seatAttendee.locale ?? "en" };
 
-    const bookingUpdated = await prisma.booking.update({
-      where: {
-        uid: reqBody.bookingUid,
-      },
-      include: {
-        attendees: true,
-      },
-      data: {
-        attendees: {
-          create: {
-            email: invitee[0].email,
-            name: invitee[0].name,
-            timeZone: invitee[0].timeZone,
-            locale: invitee[0].language.locale,
-          },
+      // See if the new date has a booking already
+      const newTimeSlotBooking = await prisma.booking.findFirst({
+        where: {
+          startTime: evt.startTime,
+          eventTypeId: eventType.id,
         },
-      },
-    });
-
-    // Add entry to bookingSeatsReference table
-    const attendeeUniqueId = uuid();
-    await prisma.bookingSeatsReferences.create({
-      data: {
-        data: {
-          description: additionalNotes,
+        select: {
+          id: true,
+          attendees: true,
         },
-        booking: {
-          connect: {
+      });
+
+      const credentials = await refreshCredentials(organizerUser.credentials);
+      const eventManager = new EventManager({ ...organizerUser, credentials });
+
+      // This is passed to the event manager to update the original calendar event
+      let originalBookingEvt;
```

Comment:
Create a `CalendarEvent` object from the original rescheduled booking. We need this to pass to the event manager to update the original calendar / video integrations